python/purehtml/ConfigFactory.py

```diff
@@ -0,0 +1,124 @@
+import logging
+
+import yaml
+from typing import Any, Dict, List, Optional, Union
+
+from purehtml.configs.Configs import Config
+from purehtml.configs.types.ArrayConfig import ArrayConfig
+from purehtml.configs.types.ConstantConfig import ConstantConfig
+from purehtml.configs.types.ObjectConfig import ObjectConfig
+from purehtml.configs.types.PrimitiveValueConfig import PrimitiveValueConfig
+from purehtml.configs.types.UnionConfig import UnionConfig
+from purehtml.transformers.TransformerFactory import TransformerFactory
+
+
+class ConfigFactory:
+    @staticmethod
+    def extract(yaml_input: Union[str, dict]) -> Config:
```

Comment:
```py
class ConfigFactory:
    @staticmethod
    def extract(yaml_input: Union[str, dict]) -> Config:
```

this is a builder function and does not **extract** anything. it could be named `fromYAML` because it `builds a new Config instance from a YAML`